### PR TITLE
Clarify note about git submodules

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Closes #n/a (See: https://www.drupal.org/node/2886531 item 1)

## Effect
PR includes the following changes:
- Add more clarity around the description of git submodules.

## Remaining Work
- [x] ACE review (@stevector)
- [x] Copy Review (@pantheon-systems/documentation)

### CODE REVIEW

**Comment by stevector** [comment]
@alexfornuto I think this change is ready to merge.

File: source/_docs/composer-drupal-8.md [hunk 1]
```diff
@@ -79,7 +79,9 @@ If Pantheon is your site's only repository, then you do have to commit the `vend
...
 
-Sometimes running Composer commands like `install`, `update`, or `require` result in the downloading of `.git` directories for the dependencies being downloaded. Downloading the Git source of a dependency may be helpful for editing and contributing back to a dependency, but it is unlikely to be needed for most site building. Use `--prefer-dist` when running `composer require` so that Composer will download a distribution of the dependency that does not include the Git history of that dependency. When you `git add` the dependency, Git will not attempt to add the dependency as [a git submodule, which is not supported on Pantheon](https://pantheon.io/docs/git-faq/#does-pantheon-support-git-submodules%3F).
+Sometimes running Composer commands like `install`, `update`, or `require` result in the downloading of `.git` directories for the dependencies being downloaded. This happens when a development (non-dist) version of a module is selected. Downloading the Git source of a dependency may be helpful for editing and contributing back to a dependency, but it is unlikely to be needed for most site building, and actually causes problems when committing these directories.
```
Comment (alexfornuto) [inline source/_docs/composer-drupal-8.md]: It looks like "non-dist" is a bit of slang, and not really a well-defined term. Is there a clearer term we can use?

**Comment by alexfornuto** [comment]
Actually, I'm going to go ahead and merge it, and we can discuss my comment as an edit after-the-fact. It's not a big enough deal to warrant holding the change back.

File: source/_docs/composer-drupal-8.md (hunk 1 above)
Comment (greg-1-anderson) [inline source/_docs/composer-drupal-8.md]: How about "development version (a release without a specific version tag)"?

There might also be a question in some people's mind on "what is a version tag", maybe.

**Pull request merged**